utils/globalSetup.js

```diff
@@ -11,28 +13,42 @@ module.exports = async function(globalConfig, projectConfig) {
     await api_module.initApi();
   }
 
-  let api = api_module.getApi();
+  const api = api_module.getApi();
 
   ipc.config.id = "nonceManager";
   ipc.config.retry = 1500;
   ipc.config.silent = false;
   ipc.config.sync = true;
   const { sudo } = utils.getEnvironmentRequiredVars();
-  const keyring = new polkadot_api.Keyring({ type: "sr25519" })
+  const keyring = new polkadot_api.Keyring({ type: "sr25519" });
   const sudoKeyringPair = keyring.createFromUri(sudo);
-  let nonce = await api.rpc.system.accountNextIndex(sudoKeyringPair.address);
-  console.info(`${nonce}`)
+  const nonce = await api.rpc.system.accountNextIndex(sudoKeyringPair.address);
+  console.info(`${nonce}`);
 
-  ipc.serve(function() {
+  ipc.serve(function () {
     ipc.server.on("getNonce", (data, socket) => {
       console.info("serving nonce" + data.id + nonce);
       ipc.server.emit(socket, "nonce-" + data.id, nonce.toNumber());
-      nonce.iaddn(1)
+      nonce.iaddn(1);
     });
   });
   ipc.server.start();
 
   // eslint-disable-next-line no-undef
   globalThis.server = ipc.server;
+  // eslint-disable-next-line no-undef
   globalThis.api = api;
+  //Setup if  debugging a gassLess test OR running gassless group
+  if (
+    (process.env.VSCODE_INSPECTOR_OPTIONS !== undefined &&
+      process.env.VSCODE_INSPECTOR_OPTIONS.length > 0 &&
+      globalConfig.testNamePattern
+        .toString()
+        .toLowerCase()
+        .includes("gassless")) ||
+    process.env.JEST_GROUP_GASSLESS
+  ) {
+    console.info("GASS LESS test - Setting it up");
+    await setupGassLess();
```

Comment:
as you can see here, if we detect that its a gassless test, we are settign it up here. So please, remove all the conditional tests